cpp/src/sampling/neighbor_sampling_impl.hpp

```diff
@@ -135,6 +137,12 @@ neighbor_sample_impl(raft::handle_t const& handle,
 
   std::vector<rmm::device_uvector<vertex_t>> level_result_src_vectors{};
   std::vector<rmm::device_uvector<vertex_t>> level_result_dst_vectors{};
+
+  // Get the number of hop. If homogeneous neighbor sample, num_edge_types = 1.
+  auto num_hops = ((fan_out.size() % num_edge_types) == 0)
+                    ? (fan_out.size() / num_edge_types)
+                    : ((fan_out.size() / num_edge_types) + 1);
+
```

Comment:
We may better define this before defining `level_result_src_vectors` and `level_result_dst_vectors`. So, we can define all `level_resutl_..._vectors` in one code block.